pkgs/development/libraries/libbacktrace/default.nix

```diff
@@ -0,0 +1,25 @@
+{ stdenv, callPackage, fetchFromGitHub, enableStatic ? false, enableShared ? true }:
+let
+  yesno = b: if b then "yes" else "no";
+in stdenv.mkDerivation rec {
+  pname = "libbacktrace";
+  version = "2018-06-05";
+  src = fetchFromGitHub {
+    owner = "ianlancetaylor";
+    repo = pname;
+    rev = "5a99ff7fed66b8ea8f09c9805c138524a7035ece";
+    sha256 = "0mb81x76k335iz3h5nqxsj4z3cz2a13i33bkhpk6iffrjz9i4dhz";
+  };
+  configureFlags = [
+    "--enable-static=${yesno enableStatic}"
+    "--enable-shared=${yesno enableShared}"
+  ];
+  passthru.static = callPackage ./. { enableStatic = true; enableShared = false; };
+  meta = with stdenv.lib; {
+    description = "A C library that may be linked into a C/C++ program to produce symbolic backtraces";
+    homepage = https://github.com/ianlancetaylor/libbacktrace;
+    author = "Ian Lance Taylor <iant@golang.org>";
+    maintainers = with maintainers; [ twey ];
+    license = with licenses; [ nonfree ];
```

Comment:
Am I missing something or isn't the license just bsd3 license?

Reply:
I'm actually not sure.  It looks *very similar* but has slight wording differences.  E.G. where BSD-3 [https://opensource.org/licenses/BSD-3-Clause](says)
>3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

`libbacktrace` instead [https://github.com/ianlancetaylor/libbacktrace/blob/master/LICENSE](has it)
```
#     (3) The name of the author may not be used to
#     endorse or promote products derived from this software without
#     specific prior written permission.
```

And libbacktrace never explicitly says that it uses BSD-3, though the gcc source, which vendors it, says:
>The libbacktrace library is provided under a BSD license. See the source files for the exact license text.

… and the text to which it refers seems identical to the contents of this file.

Reply:
It seems like a minor difference that results in almost the same effect, I would say it's ok but I'm not a lawyer. The difference is that this license doesn't mention contributors' names as names that cannot be used to endorse derived software without permission. But "author" vs "copyright holder" is probably the same :woman_shrugging: 

> And libbacktrace never explicitly says that it uses BSD-3, though the GCC source, which vendors it, says: [...]

Well I saw that the readme for https://github.com/ianlancetaylor/libbacktrace says that as well, or is that already what you meant? Is that repository not the canonical source? Because the c source files also seem to have the same bsd3-like license header.

Reply:
Yeah, I think you're right and it is BSD-3 in some form.  The licensing info here is only informational anyway.  I've updated it accordingly.

Reply:
If a license is set to unfree then users need to explicitly opt in to allow nonfree software, I wouldn't say it's just informational. Thanks :+1:

Reply:
I mean that it's informational in the sense that if we say it's BSD-3 and it turns out not to be BSD-3, we aren't in legal trouble.  :)  So it's probably safe to mark it as BSD-3.

Reply:
Ah right makes sense.